frontend/src/employee-frontend/components/vasu/templates/CreateQuestionModal.tsx

```diff
@@ -0,0 +1,141 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import React, { useState } from 'react'
+import FormModal from '../../../../lib-components/molecules/modals/FormModal'
+import {
+  VasuQuestion,
+  VasuQuestionType,
+  VasuQuestionTypes
+} from '../../../api/child/vasu'
+import InputField from '../../../../lib-components/atoms/form/InputField'
+import { Label } from 'lib-components/typography'
+import { useTranslation } from '../../../state/i18n'
+import {
+  FixedSpaceColumn,
+  FixedSpaceRow
+} from 'lib-components/layout/flex-helpers'
+import Combobox from '../../../../lib-components/atoms/form/Combobox'
+import IconButton from '../../../../lib-components/atoms/buttons/IconButton'
+import { faTrash } from '../../../../lib-icons'
+import InlineButton from '../../../../lib-components/atoms/buttons/InlineButton'
+
+interface Props {
+  onSave: (question: VasuQuestion) => void
+  onCancel: () => void
+}
+
+export default React.memo(function CreateQuestionModal({
+  onCancel,
+  onSave
+}: Props) {
+  const { i18n } = useTranslation()
+  const [type, setType] = useState<VasuQuestionType>('TEXT')
+  const [name, setName] = useState('')
+  const [options, setOptions] = useState([''])
+
+  function createQuestion(): VasuQuestion {
+    switch (type) {
+      case 'TEXT':
+        return {
+          type: 'TEXT',
+          name: name,
+          multiline: false,
+          value: ''
+        }
+      case 'CHECKBOX':
+        return {
+          type: 'CHECKBOX',
+          name: name,
+          value: false
+        }
+      case 'RADIO_GROUP':
+        return {
+          type: 'RADIO_GROUP',
+          name: name,
+          optionNames: options,
+          value: null
+        }
+      case 'MULTISELECT':
+        return {
+          type: 'MULTISELECT',
+          name: name,
+          optionNames: options,
+          minSelections: 0,
+          maxSelections: null,
+          value: []
+        }
+      default:
+        throw Error('Unexpected type')
+    }
```

Comment:
I don't know how our typescript compiler / linter is configured, but I prefer switch without default case and handling all cases explicitly and throwing after the switch. This way the compiler might warn about an unhandled case when a new type is added.

Reply:
ah, didn't work without default before but adding `as const` to the enum helped